custom/enikshay/private_sector_datamigration/factory.py

```diff
@@ -354,3 +361,45 @@ def _location_owner(self):
     @memoized
     def _location_owner_id(self):
         return self._location_owner.location_id
+
+    @property
+    @memoized
+    def _virtual_user(self):
+        return CommCareUser.get(self._location_owner.user_id)
+
+    @property
+    @memoized
+    def _id_issuer_number(self):
+        return self._virtual_user.user_data['id_issuer_number']
+
+    @property
+    @memoized
+    def _id_issuer_body(self):
+        return self._virtual_user.user_data['id_issuer_body']
+
+    @property
+    def _id_device_body(self):
+        return compress_nikshay_id(0, 0)
+
+    @property
+    @memoized
+    def _serial_count(self):
+        return MigratedBeneficiaryCounter.get_next_counter()
+
+    @property
+    @memoized
+    def _serial_count_compressed(self):
+        return compress_nikshay_id(self._serial_count, 2)
+
+    @property
+    @memoized
+    def person_id_flat(self):
+        return self._id_issuer_body + self._id_device_body + self._serial_count_compressed
+
+    @property
+    def person_id(self):
```

Comment:
Looks like this will turn `"abcdefg"` into `"abc-def-g"`, is that okay, or should it be `"a-bcd-efg"`